esp/platform/espcfg.cpp

```diff
@@ -111,6 +111,73 @@ StringBuffer &CVSBuildToEspVersion(char const * tag, StringBuffer & out)
     return out;
 }
 
+void CEspConfig::ensureSDSSessionDomains()
+{
+    bool hasDefaultSessionDomain = false;
+    Owned<IPropertyTree> proc_cfg = getProcessConfig(m_envpt, m_process.str());
+    Owned<IPropertyTreeIterator> it = proc_cfg->getElements("AuthDomains/AuthDomain");
+    ForEach(*it)
+    {
+        IPropertyTree& authDomain = it->query();
+        const char* authType = authDomain.queryProp("@authType");
+        if (isEmptyString(authType) || (!strieq(authType, "AuthPerSessionOnly") && !strieq(authType, "AuthTypeMixed")))
+            continue;
+
+        const char* authDomainName = authDomain.queryProp("@name");
+        if (isEmptyString(authDomainName))
+        {
+            if (hasDefaultSessionDomain)
+                throw MakeStringException(-1, ">1 AuthDomains are not named.");
+
+            hasDefaultSessionDomain = true;
+            authDomainName = "default";
+        }
+
+        Owned<IRemoteConnection> conn = querySDS().connect(PathSessionRoot, myProcessSession(), RTM_LOCK_WRITE, SESSION_SDS_LOCK_TIMEOUT);
+        if (!conn)
+        {
+            conn.setown(querySDS().connect("/", myProcessSession(), RTM_LOCK_WRITE|RTM_CREATE_QUERY, SESSION_SDS_LOCK_TIMEOUT));
```

Comment:
This is not what I meant by my previous comment: https://github.com/hpcc-systems/HPCC-Platform/pull/9999#discussion_r121642984

(you can never CREATE "/")


What I was suggesting is that you use RTM_CREATE_QUERY like this:
```
    Owned<IRemoteConnection> conn = querySDS().connect(PathSessionRoot, myProcessSession(), RTM_CREATE_QUERY|RTM_LOCK_WRITE, SESSION_SDS_LOCK_TIMEOUT);
```

that will create PathSessionRoot is missing, so you don't need 2 connects or the ensurePTree.